sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/firestore/FirestoreV1FnRunQueryTest.java

```diff
@@ -283,62 +299,86 @@ protected RunQueryFn getFn(
 
   private static final class TestData {
 
+    static final FieldReference FILTER_FIELD_PATH =
+        FieldReference.newBuilder().setFieldPath("foo").build();
+    static final Filter FIELD_EQUALS_BAR =
+        Filter.newBuilder()
+            .setFieldFilter(
+                FieldFilter.newBuilder()
+                    .setField(FILTER_FIELD_PATH)
+                    .setOp(Operator.EQUAL)
+                    .setValue(Value.newBuilder().setStringValue("bar"))
+                    .build())
+            .build();
+    static final Filter FIELD_NOT_EQUALS_FOO =
+        Filter.newBuilder()
+            .setFieldFilter(
+                FieldFilter.newBuilder()
+                    .setField(FILTER_FIELD_PATH)
+                    .setOp(Operator.NOT_EQUAL)
+                    .setValue(Value.newBuilder().setStringValue("foo"))
+                    .build())
+            .build();
+
     private final RunQueryRequest request;
     private final RunQueryResponse response1;
     private final RunQueryResponse response2;
     private final RunQueryResponse response3;
 
-    public TestData(String projectId, Function<FieldReference, List<Order>> orderFunction) {
-      String fieldPath = "foo";
-      FieldReference foo = FieldReference.newBuilder().setFieldPath(fieldPath).build();
+    public TestData(
+        String projectId, Function<FieldReference, List<Order>> orderFunction, Filter filter) {
       StructuredQuery.Builder builder =
           StructuredQuery.newBuilder()
               .addFrom(
                   CollectionSelector.newBuilder()
                       .setAllDescendants(false)
                       .setCollectionId("collection"))
-              .setWhere(
-                  Filter.newBuilder()
-                      .setFieldFilter(
-                          FieldFilter.newBuilder()
-                              .setField(foo)
-                              .setOp(Operator.EQUAL)
-                              .setValue(Value.newBuilder().setStringValue("bar"))
-                              .build()));
-
-      orderFunction.apply(foo).forEach(builder::addOrderBy);
+              .setWhere(filter);
+
+      orderFunction.apply(FILTER_FIELD_PATH).forEach(builder::addOrderBy);
       request =
           RunQueryRequest.newBuilder()
               .setParent(String.format("projects/%s/databases/(default)/document", projectId))
               .setStructuredQuery(builder)
               .build();
 
-      response1 = newResponse(fieldPath, 1);
-      response2 = newResponse(fieldPath, 2);
-      response3 = newResponse(fieldPath, 3);
+      response1 = newResponse(1);
+      response2 = newResponse(2);
+      response3 = newResponse(3);
     }
 
-    private static RunQueryResponse newResponse(String field, int docNumber) {
+    /**
+     * Returns single-document response like this: { "__name__": "doc-{docNumber}", "foo": "bar",
+     * "fo`o.m`ap": { "bar.key": "bar.val" } }
```

Comment:
You added a super complicated case "\`fo\\\\\`o.m\\\\\`ap\`.\`bar.key\`" and the most simple case "foo" for the order by, can we also add two test cases that are in between these two: (1) x.y as y inside the map x, both x and y are simple strings and not escaped; (2) \`x.y\` as a single property with a dot in the property name. So the document might look like this:
```
{
  "foo": "bar"
  "fo`o.m`ap": { "bar.key": "bar.val" }
  "baz" : {"qux" : "val"}
  "quux.quuz" : 123
}
```
you might parameterize the tests if it makes things easier, or just writing these 4 separate tests also seem fine.

Reply:
Done.